crates/serde-lib/src/bcs_safe_btreeset.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+/// Cryptographic applications using BCS cannot use BTreeSet as it does not enforce canonicity.
```

Comment:
It's not necessarily only cryptography, determinism is the feature we're looking here and it might also be useful for consistent logs and similar applications were serialization determinism is important.